proposals/static-delegates.md

```diff
@@ -0,0 +1,74 @@
+# Static Delegates
+
+* [x] Proposed
+* [ ] Prototype: Not Started
+* [ ] Implementation: Not Started
+* [ ] Specification: Not Started
+
+## Summary
+[summary]: #summary
+
+Provide a general-purpose, lightweight callback capability to the C# language.
+
+## Motivation
+[motivation]: #motivation
+
+Today, users have the ability to create callbacks using the `System.Delegate` type. However, these are fairly heavyweight (such as requiring a heap allocation and always having handling for chaining callbacks together).
+
+Additionally, `System.Delegate` does not provide the best interop with unmanaged function pointers, namely due being non-blittable and requiring marshalling anytime it crosses the managed/unmanaged boundary.
+
+With a few minor tweaks, we could provide a new type of delegate that is lightweight, general-purpose, and interops well with native code.
+
+## Detailed design
+[design]: #detailed-design
+
+One would declare a static delegate via the following:
+
+```C#
+static delegate int Func()
+```
+
+One could additionally attribute the declaration with something similar to `System.Runtime.InteropServices.UnmanagedFunctionPointer` so that the calling convention, string marshalling, and set last error behavior can be controlled. NOTE: Using `System.Runtime.InteropServices.UnmanagedFunctionPointer` itself will not work, as it is only usable on Delegates.
+
+The declaration would get translated into an internal representation by the compiler that is similar to the following
+
+```C#
+struct <name>
```

Comment:
I put a placeholder here because I think just `Func` isn't enough, it would need to be a non-user constructable identifier as we do with other backing structs. For example, with fixed size buffers we do `<name>e__FixedBuffer` (the `<` and `>` are not placeholders but part of the actual identifier), so in this case we would do something like `<Func>e__StaticDelegate` (or something similar, I would imagine).